app/javascript/hooks/forms/users/authentication/useSignInForm.js

```diff
@@ -0,0 +1,68 @@
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useTranslation } from 'react-i18next';
+import { useForm } from 'react-hook-form';
+import { useCallback, useMemo } from 'react';
+
+export function useSignInFormValidation() {
+  return useMemo(() => (yup.object({
+    email: yup.string().required('forms.validations.email.required').email('forms.validations.email.email'),
+    password: yup.string().required('forms.validations.password.required'),
+    extend_session: yup.bool(''),
+  })), []);
+}
+
+export default function useSignInForm({ _defaultValues, ..._config } = {}) {
+  const { t, i18n } = useTranslation();
+
+  const fields = useMemo(() => ({
+    email: {
+      label: t('forms.user.signin.fields.email.label'),
+      placeHolder: t('forms.user.signin.fields.email.placeholder'),
+      controlId: 'signInFormEmail',
+      hookForm: {
+        id: 'email',
+      },
+    },
+    password: {
+      label: t('forms.user.signin.fields.password.label'),
+      placeHolder: t('forms.user.signin.fields.password.placeholder'),
+      controlId: 'signInFormPwd',
+      hookForm: {
+        id: 'password',
+      },
+    },
+    extend_session: {
+      label: t('forms.user.signin.fields.remember_me.label'),
+      controlId: 'signInFormRememberMe',
+      hookForm: {
+        id: 'extend_session',
+      },
+    },
+  }), [i18n.resolvedLanguage]);
+
+  const validationSchema = useSignInFormValidation();
+
+  const config = useMemo(() => ({
+    ...{
+      mode: 'onSubmit',
+      criteriaMode: 'firstError',
+      defaultValues: {
+        ...{
+          email: '',
+          password: '',
+          extend_session: false,
+        },
+        ..._defaultValues,
+      },
+      resolver: yupResolver(validationSchema),
+    },
+    ..._config,
```

Comment:
When configuring the form and providing the default values.
Sometimes we may need to inject more configuration externally.
The idea was to merge the local config with the injected config prioritizing the latter.
However, that will not work since the react-hook-form config object is multi dimensional (or multi level).
The defaultValues prop is an object with another level deep.
So shallow merging the defaultValues will overwrite it meaning that to extend one default value, one is required to redefine all of the others (which means being aware of the underlying implementation of the form hook that removes its usage).
The solution is to deeply merge the objects which is complicated to do or to use through a lib,
Since the react-hook-form API confirms that only the defaultValues prop is an object.
The idea was to extract it and merge it independently therefore making a deep merge with two levels of shallow merge.